src/Data/Binary/Serialise/CBOR/Tutorial.hs

```diff
@@ -16,33 +16,159 @@ module Data.Binary.Serialise.CBOR.Tutorial
     -- ** The CBOR format
     -- $cborformat
 
+    -- ** The 'Serialise' class
+    -- $serialise
+
     -- ** Encoding terms
     -- $encoding
 
     -- ** Decoding terms
     -- $decoding
   ) where
 
+import Data.Binary.Serialise.CBOR
+
 {- $introduction
 
-Lorem ipsum...
+@binary-serialise-cbor@ is a library for the serialisation of Haskell values.
+
+As in modern modern serialisation libraries, @binary-serialise-cbor@ offers
+instance derivation via GHC's 'GHC.Generic' mechanism,
+
+> import qualified Data.Binary.Serialise.CBOR as CBOR
+> import qualified Data.ByteString.Lazy as BSL
+>
+> data Animal = HoppingAnimal { animalName :: String, hoppingHeight :: Int }
+>             | WalkingAnimal { animalName :: String, walkingSpeed :: Int }
+>             deriving (Generic)
+>
+> instance CBOR.Serialise Animal
+>
+> fredTheFrog :: Animal
+> fredTheFrog = HoppingAnimal "Fred" 4
+>
+> main = BSL.writeFile "hi" (CBOR.serialise fredTheFrog)
+
+We can then later read Fred,
+
+> main = do
+>     fred <- CBOR.deserialise <$> BSL.readFile "hi"
+>     print fred
 
 -}
 
 {- $cborformat
 
-Lorem ipsum...
+@binary-serialise-cbor@ uses the Concise Binary Object Representation, CBOR
+(IETF RFC 7049), as its serialised representation. This encoding is efficient in
+both encoding\/decoding complexity as well as space, and is generally
+machine-independent.
+
+The CBOR data model resembles that of JSON, having arrays, key/value maps,
+integers, floating point numbers, and a variety of string types. In addition,
+CBOR allows items to be /tagged/ with a number which describes the type of data
```

Comment:
This is somewhat confusing. CBOR tagging is for CBOR extensions, the constructor tagging is entirely an encoding scheme that we use and is not something you'll find in the CBOR spec.